exa/core/data.py

```diff
@@ -228,11 +228,13 @@ def _set_categories(self, df, reverse=False):
                 )
         return df
 
-    def __init__(self, *args, df=None, **kws):
+    def __init__(self, *args, source=pd.read_csv, df=None, **kws):
```

Comment:
`source` can be forwarded directly in the `super` call so we need not set it explicitly.

I don't think we want to mutate the constructor to `Data` to the point of optionally accepting positional args which are all re-routed to `self.source(*args, **kwargs)` but we want to support lazy evaluation of the callable.

I originally intended to support this pattern with `call_args` and `call_kws` attributes where args you *would* pass to the source callable are separated from args to `Data` itself, but perhaps that could be tedious and confusing for new users.

This hints back to @avmarchenko and his original comment about a classmethod, e.g.
```python
class Data:
    @classmethod
    def from_csv(cls, *args, **kwargs):
        reurn cls(call_args=args, source=pd.read_csv, call_kws=kwargs)
```

Then we don't have to "default" support pd.read_csv for the unsuspecting user, rather the classmethod provisions the instance of `Data` according to the assumptions about the source of the data.

The opposite problem of how to provide args and kwargs to the constructor of `Data` from the classmethod is maybe less relevant, as the API is informing the user of a convenient supported use case, and if one required specifying both, it is still entirely possible from the `Data.__init__`